linera-views/src/test_utils/performance.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    fmt::Debug,
+    time::{Duration, Instant},
+};
+
+use crate::{
+    batch::Batch,
+    store::LocalKeyValueStore,
+    test_utils::{add_prefix, get_random_key_values2},
+};
+
+// We generate about 2000 keys of length 11 with a key of length 10000
+// The keys are of the form 0,x_1, ..., x_n with 0 <= x_i < 4 and n=10.
+
+/// A value to use for the keys
+const PREFIX: &[u8] = &[0];
+
+/// A value to use for the keys
+const PREFIX_SEARCH: &[u8] = &[0, 0];
+
+/// The number of keys
+const NUM_ENTRIES: usize = 200;
+
+/// The number of inserted keys
+const NUM_INSERT: usize = 70;
+
+/// The length of the keys
+const LEN_KEY: usize = 10;
+
+/// The length of the values
+const LEN_VALUE: usize = 10000;
+
+async fn clear_store<S: LocalKeyValueStore>(store: &S) {
+    let mut batch = Batch::new();
+    batch.delete_key_prefix(PREFIX.to_vec());
+    store.write_batch(batch).await.unwrap();
+}
+
+/// Compute benchmark of the `contains_key` operation.
+pub async fn performance_contains_key<S: LocalKeyValueStore, F>(
+    store: S,
+    iterations: u64,
+    f: F,
```

Comment:
If this is always meant to be `black_box`, I'd call it `black_box` instead of `f`.

But actually, I'd prefer just using `criterion` here, and putting the whole module behind a feature flag. Then applications that don't specify that flag won't depend on criterion.

Reply:
Right now, it is only `black_box`, but conceivably something else could be used.

The problem here is that this forces having the `criterion` dependency in the `Cargo.lock` file of examples, something that we do not want. I prefer that to have a feature flag.

Reply:
Example of other usage for the function `f`:
* A blank function, which is not a black box. Would be interesting to know what the compiler do.
* Processing the `S::Keys` and `S::KeyValues` and see what happens.